terraform/context.go

```diff
@@ -92,6 +93,24 @@ func NewContext(opts *ContextOpts) *Context {
 		state.init()
 	}
 
+	// If our state is from the future, then error. Callers can avoid
+	// this error by explicitly setting `StateFutureAllowed`.
+	if !opts.StateFutureAllowed && state.FromFutureTerraform() {
+		return nil, fmt.Errorf(
+			"Terraform doesn't allow running any operations against a state\n"+
+				"that was written by a future Terraform version. The state is\n"+
+				"reporting it is written by Terraform '%s'.\n\n"+
+				"Please run at least that version of Terraform to continue. Many\n"+
+				"command line operations also provide a flag to override this\n"+
+				"protection",
```

Comment:
Based on the PR description, this last sentence is not accurate. Sounds like we're going for manual edit-the-state override in this case?

If this is the case, I wonder if the thing to do here is add a docs page and link off to it in the error message?